test/oxo.js

```diff
@@ -0,0 +1,164 @@
+        .then(() => oxoInst.move(1,2,{from: player1}))
+            .then(() => oxoInst.movesCount())
+            .then(move => assert.equal(Number(move)-1, 3, 'err move1'))
+
+    });
+
+    it('should fail if player2 move first, on move', () => {
+        return Promise.resolve()
+            .then(() => oxoInst.joinGame({from: player1, value: bets}))
+            .then(() => oxoInst.joinGame({from: player2, value: bets}))
+            .then(() => asserts.throws(oxoInst.move(2,2,{from: player2})));
+    });
+
+    it('should fail if move in occupied cell', () =>{
+        return Promise.resolve()
+            .then(() => oxoInst.joinGame({from: player1, value: bets}))
+            .then(() => oxoInst.joinGame({from: player2, value: bets}))
+            .then(() => oxoInst.move(1,1,{from: player1}))
+            .then(() => asserts.throws(oxoInst.move(1,1,{from: player2})));
+    });
+
+    it('should emit Move event on move', () => {
+
+        return Promise.resolve()
+            .then(() => oxoInst.joinGame({from: player1, value: bets}))
+            .then(() => oxoInst.joinGame({from: player2, value: bets}))
+            .then(() => oxoInst.move(2,1,{from: player1}))
+            .then(result => {
+                assert.equal(result.logs.length, 1);
+                assert.equal(result.logs[0].event, 'Move');
+                assert.equal(result.logs[0].args.col, "2");
+                assert.equal(result.logs[0].args.row, "1");
+            });
+    });
+
+//////////////////// checkWiner//////////////////////////
+
+            it('check win on checkWiner', () => {
+
+                return Promise.resolve()
+                    .then(() => oxoInst.joinGame({from: player1,
+                        value: bets}))
+                    .then(() => oxoInst.joinGame({from: player2,
+                        value: bets}))
+                    .then(() => oxoInst.move(0, 0, {from: player1}))
+                    .then(() => oxoInst.move(1, 0, {from: player2}))
+
+                    .then(() => oxoInst.move(0, 1, {from: player1}))
+                    .then(() => oxoInst.move(2, 0, {from: player2}))
+                    .then(() => oxoInst.move(0, 2, {from: player1}))
+
+
+                    .then(result => {
+                        assert.equal(result.logs.length, 4);
+                        assert.equal(result.logs[1].event, 'Win');
+                        assert.equal(result.logs[1].args.addr, player1);
+                        assert.equal(result.logs[1].args.player
+                            .toString(), "1");
+                        assert.equal(result.logs[1].args.gain.toString(), (bets*2*90/100/ether).toString());
+
+                    });
+
+            });
+
+
+//      Остальные тесты дожму на следующей неделе, 26-27 марта
+// уже после блокчейн юа хакатона
+//        (всю неделю сплошные конфы и хакатоны, на прошлых выходных токен на кулс хакатоне пилили)
+//
+//    Доделать:
+// проверки трансферов, ничья, обнуление игры
```

Comment:
What will happen if I'll try to make join game twice? 
Is first player able to move if second player isn't exist?
Is some player able to move after game's ended? 
What if someone try to move twice in a turn?
I suggest to add tests for this too 🐱